### PR TITLE
MM-33103 Generate Support Packet Self-Managed

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://mattermost.atlassian.net/browse/MM-33103

- Created new common Support Packet file to be referenced and included by both Cloud and Self-Managed

### CODE REVIEW

**Comment by cwarnermm** [comment]
@amyblais While these changes would traditionally be applied to the v5.33-documentation branch, they have been made on a branch off master instead to replicate our future process of PRs against the Staging Environment, and to develop docs in a Cloud-first way.

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:14]:
```suggestion
- ``mattermost.log``
```

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:15]:
```suggestion
- ``plugins.json``
```

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:16]:
```suggestion
- ``sanitized_config.json``
```

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:17]:
```suggestion
- ``support_packet.yaml``
```

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:18]:
```suggestion
- ``warning.txt``
```

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:20]:
```suggestion
You'll be notified if any packet files are unavailable during packet generation. See the ``warning.txt`` file for details.
```

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:5]:
```suggestion
  System Admins should always sanitize the contents to remove sensitive information before sending the support packet to Mattermost by email or via a support ticket.
```

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:22]: We mention that passwords are automatically stripped in the opening paragraph, but mention here that they should be removed. I'm also wondering whether there's a specific method we recommend in terms of removing this information, or if it's something a System Admin would be inherently familiar with?

**Review by justinegeffen — COMMENTED** [review]
This is great, thanks! Some comments around the wording of step three but otherwise looks like single-sourcing is a win. :)

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (cwarnermm) [reply source/cloud/cloud-administration/common-support-packet.rst:22]: @justinegeffen This is a great set of questions worth exploring with R&D further. If we can provide thought leadership and guidance to System Admins, they can complete these tasks faster and with greater confidence. 

@hahmadia  - Can you clarify what specifically gets sanitized when the support packet is generated, and what we recommend System Admins should do to further sanitize manually prior to sending the support packet to Mattermost? Thanks!

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (hahmadia) [inline source/cloud/cloud-administration/common-support-packet.rst:1]: Not sure if important to mention but it includes plugins installed too. I'm not sure if it falls under `external dependencies`.

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (hahmadia) [inline source/cloud/cloud-administration/common-support-packet.rst:18]: `warning.txt` only appears if one of the files failed to be retrieved for whatever reason. It won't always be there if the above 4 files are retrieved without problem. Again not sure if it's important to point it out.

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (hahmadia) [reply source/cloud/cloud-administration/common-support-packet.rst:22]: Sure thing! 

Nothing in `mattermost.log` , `plugins.json`, `support_packet` or `warning.txt` get sanitized or removed.

The following values in the `config.json` get sanitized:

- LdapSettings.BindPassword
- FileSettings.PublicLinkSalt
- FileSettings.AmazonS3SecretAccessKey
- EmailSettings.SMTPPassword
- GitLabSettings.Secret
- GoogleSettings.Secret 
- Office365Settings.Secret
- OpenIdSettings.Secret 
- SqlSettings.DataSource
- SqlSettings.AtRestEncryptKey 
- ElasticsearchSettings.Password
- All SqlSettings.DataSourceReplicas
- All SqlSettings.DataSourceSearchReplicas
- MessageExportSettings.GlobalRelaySettings.SmtpPassword
- ServiceSettings.GfycatApiSecret 
- ServiceSettings.SplitKey

If they like and feel this isn't enough, they can sanitize / remove whatever other data they want.

The  LDAP groups are not generated for us to view. Only LDAP Version and LDAP Vendor are generated if they are available in the `support_packet.yaml`

As for admins doing anything further, I think that's not required. I'd just recommend them to skim through the `plugins.json` and ensure their plugins don't have any sort of confidential information that needs to be hidden (like secret keys) because plugins are not sanitized but other than that I think they should be okay.

**Review by hahmadia — APPROVED** [review]
LGTM! Thanks! Just a few minor suggestions but nothing major 👍

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (cwarnermm) [reply source/cloud/cloud-administration/common-support-packet.rst]: Thank you for the great details, @hahmadia! I've requested a second review from you to ensure that I've captured these details accurately in the product documentation.

**Review by hahmadia — APPROVED** [review]
Yep! That sounds right to me! Thank you

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:18]:
```suggestion
- ``warning.txt`` (present when issues are encountered during packet generation)
```

**Review by justinegeffen — APPROVED** [review]

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:7]:
```suggestion
To generate a Support Packet:
```

File: source/cloud/cloud-administration/common-support-packet.rst
Comment (justinegeffen) [inline source/cloud/cloud-administration/common-support-packet.rst:26]:
```suggestion
  When sanitizing Support Packet data, replace details with example strings that contain the same special characters if possible, as special characters are common causes of configuration errors.
```

**Review by adamjclarkson — APPROVED** [review]

**Comment by amyblais** [comment]
@cwarnermm This and https://github.com/mattermost/docs/pull/4382 have merge conflicts. I'd rather not publish the v5.33 release blog post until this is resolved and these docs are merged.

**Pull request merged**
